File: client/src/communication/bet_content.go
package communication

import (
	"fmt"
	"strings"
)

type BetContent struct {
	Name      string
	LastName  string
	IdNumber  string
	Birthdate string
	BetNumber string
}

func (bc *(BetContent)) Serialize() string {
	content := fmt.Sprintf("%s, %s, %s, %s, %s", bc.Name, bc.LastName, bc.IdNumber, bc.Birthdate, bc.BetNumber)
	return content
}

func DeserializeBetContent(content string) *BetContent {
	fields := strings.Split(content, ",")
	name := fields[0]
	lastName := fields[1]
	idNumber := fields[2]
	birthdate := fields[3]
	betNumber := fields[4]
	return &BetContent{Name: name, LastName: lastName, IdNumber: idNumber, Birthdate: birthdate, BetNumber: betNumber}
}

File: server/src/communication/bet_content.py
class BetContent:
    _valueSeparator = ", "
    
    def __init__(self, name, lastname, id_number, birthdate, bet_number):
        self.name = name
        self.lastname = lastname
        self.id_number = id_number
        self.birthdate = birthdate
        self.bet_number = bet_number

    def deserialize(serializedBetContent):
        name, lastname, id_number, birthdate, bet_number = serializedBetContent.split(BetContent._valueSeparator)
        return BetContent(name, lastname, id_number, birthdate, bet_number)
